src/librustdoc/html/render/mod.rs

```diff
@@ -1451,27 +1449,19 @@ pub(crate) fn notable_traits_button(ty: &clean::Type, cx: &Context<'_>) -> Optio
         return None;
     }
 
-    if let Some(impls) = cx.cache().impls.get(&did) {
-        for i in impls {
-            let impl_ = i.inner_impl();
-            if impl_.polarity != ty::ImplPolarity::Positive {
-                continue;
-            }
-
-            if !ty.is_doc_subtype_of(&impl_.for_, cx.cache()) {
-                // Two different types might have the same did,
-                // without actually being the same.
-                continue;
-            }
-            if let Some(trait_) = &impl_.trait_ {
-                let trait_did = trait_.def_id();
-
-                if cx.cache().traits.get(&trait_did).is_some_and(|t| t.is_notable_trait(cx.tcx())) {
-                    has_notable_trait = true;
-                }
-            }
-        }
-    }
+    let impls = cx.cache().impls.get(&did)?;
+    let has_notable_trait = impls
+        .iter()
+        .map(Impl::inner_impl)
+        .filter(|impl_| impl_.polarity == ty::ImplPolarity::Positive)
+        .filter(|impl_| {
+            // Two different types might have the same did,
+            // without actually being the same.
+            ty.is_doc_subtype_of(&impl_.for_, cx.cache())
+        })
```

Comment:
```suggestion
        .filter(|impl_| {
            impl_.polarity == ty::ImplPolarity::Positive &&
            // Two different types might have the same did,
            // without actually being the same.
            ty.is_doc_subtype_of(&impl_.for_, cx.cache())
        })
```

Reply:
Done. BTW, this was what I did initially, but switch to two `.filter()`s because `rustfmt` doesn't handle comments between two parts of a binary op, but I agree that this is a better way to write it.